sql/2019/08_Security/08_35.sql

```diff
@@ -1,23 +1,35 @@
 #standardSQL
```

Comment:
Looks good, but when I run this in BQ I'm getting different results than what you have in the google doc

| client | http_only | total	 | pct   |
|--------|-----------|---------|-------|
| desktop| 467463    | 4371973 | 10.69 |
| mobile | 520748    | 5297442 | 9.83  |

Reply:
Really good catch.

After some debugging, this has to do with `return header.split(':')[1].trim();`. Header values may have a `:` in their value (like an expires directive) and this would be cutting off the return value too short.

Many other metrics use this splitting approach, so I'll have to go through them all with a fix.

The reason for the difference in results is because earlier I had used `return header.split(': ')[1]` which included a space after the colon, and was less likely to split up things like expires directives, which have numbers after it. I updated all queries to use the trim approach, assuming it wouldn't affect the results.

A proper fix is to take a substring after the first colon rather than splitting on any colons.


Thanks for catching this!!